cmd/wallet-sdk-gomobile/did/creator/creator.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package creator is a gomobile-compatible version of github.com/trustbloc/wallet-sdk/pkg/did/creator.
+// It takes care of the necessary wrapping/conversion between the Go SDK and the gomobile sdk
+package creator
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/api"
+	goapi "github.com/trustbloc/wallet-sdk/pkg/api"
+	goapididcreator "github.com/trustbloc/wallet-sdk/pkg/did/creator"
+)
+
+const ed25519VerificationKey2018 = "Ed25519VerificationKey2018"
+
+// DIDCreator is used for creating DID Documents using supported DID methods (currently only did:key).
+type DIDCreator struct {
+	goAPIDIDCreator *goapididcreator.DIDCreator
+}
+
+// NewDIDCreator returns a new DIDCreator.
+func NewDIDCreator() *DIDCreator {
+	return &DIDCreator{goAPIDIDCreator: goapididcreator.NewDIDCreator()}
+}
+
+// Create creates a DID document for the given DID method.
+// The usage of createDIDOpts depends on the DID method you're using.
+func (d *DIDCreator) Create(method string, createDIDOpts *api.CreateDIDOpts) ([]byte, error) {
```

Comment:
We should discuss how the caller is expected to pass their key in. In this PR, I have them pass in the public key directly, assuming they generated it somehow (maybe with the KMS implementation, to be done in a future PR?)

Reply:
Updated based on a discussion earlier